File: app/src/main/java/com/liusbl/listbinderssample/multi/PersonListItem.java
package com.liusbl.listbinderssample.multi;

public class PersonListItem extends PersonSubListItem {
    private final String name;

    public PersonListItem(String id, String name) {
        super("PersonListType.PERSON");
        this.name = name;
    }

    @Override
    public Enum getViewType() {
        return PersonListType.PERSON;
    }

    public String getName() {
        return name;
    }
}

File: app/src/main/java/com/liusbl/listbinderssample/multi/PersonListItem.kt
package com.liusbl.listbinderssample.multi

import com.liusbl.listbinders.ListItem

sealed class PersonListItem(id: String?, type: ListType) : ListItem(id, type) {
    data class Person(
        val id: String,
        val name: String
    ) : PersonListItem(id, ListType.PERSON)

    data class Header(
        val title: String
    ) : PersonListItem(title, ListType.HEADER)

    object Footer : PersonListItem(null, ListType.FOOTER)

    enum class ListType {
        PERSON,
        HEADER,
        FOOTER
    }
}